service_config.md

```diff
@@ -0,0 +1,194 @@
+ |  |   snippet2: |                      |  | handlers=stdout                  |
+ |  | 	##################               |  | ##################               |
+ |  | 	# Log Formatters #               |  | # Log Formatters #               |
+ |  | 	##################               |  | ##################               |
+ |  | 	[formatter_normal]               |  | [formatter_normal]               |
+ |  | 	format=(%(name)s): s%(message)s  |  | format=(%(name)s): s%(message)s  |
+ |  +------------------------------------+  | [formatters]                     |
+ |                                          | keys=normal                      |
++-------------------------+                 +----------------------------------+
+| apiVersion: v1          |                           |
+| kind: Secret            |                           |
+| metadata:               |                           |
+|   name: service-secret2 |                           |
+|   namespace: openstack  |        1. Mounted as Volume by Service Pod
+| stringData:             |
+|   snippet3: |           |        2. `kolla_set_configs && kolla_start`: sync
+|  [formatters]           |           the resulting secret as done for the other
+|  keys=normal            |           regular config files
++-------------------------+
+```
+
+As a result of this strategy, the service presents a layout similar to the
+following:
+
+```
+00-default.conf            => default configs generated by operator. This is stored in ConfigMap
+01-deployment-secrets.conf => default configs generated by operator, which contains credentials such as [database] connection . This is stored in Secret
+02-global.conf             => custom configs provided by users via top-level customServiceConfig. This is stored in ConfigMap
+03-service.conf            => custom configs provided by users via service level customServiceConfig. This is stored in ConfigMap
+04-secrets.conf            => custom configs provided by users via service level customServiceConfigSecrets, which contains credentials. This is stored in Secret and would not be present if no secrets are provided
+```
+
+
+The service will pass the `--config-dir` parameter to point to the
+`<service>.conf.d` directory where all the config files listed above are
+rendered.
+
+```
+"command": "/usr/bin/<service> --config-dir /etc/<service>/<service>.conf.d"
+```
+
+If this strategy is not available, the `init container` that executes a start
+script (e.g., `init.sh`) won't be removed, and the logic that generates the
+layout mentioned above will be implemented in the init script.
+
+An example of the `customServiceConfigSecret` usage can be found in Manila,
+where this parameter has been used to test the
+[NetApp backend](https://gist.github.com/gouthampacha/1b5681104ee066b5bd2c702b29376199)
+
+
+## Conclusion
+
+The model described here allows to reach many goals:
+- remove (entirely) the `initContainer` and the related scripts that are no
```

Comment:
as from my comment above there might be still a use case for init container, if e.g. IP information is required to setup the service/service-config.

Reply:
+1, I'm going to change the wording here, and as per your suggestion in the previous comment I'm going to state that there might be cases where we'd like to keep the init container to properly handle the service start (e.g. you need to know the IP address on the Multus network, which is known when the Pod is created and started).